_posts/2024-07-22-developer-update.md

```diff
@@ -0,0 +1,115 @@
+
+This post highlights development activities that are **available to be worked on today**, along with interested [developers](https://github.com/geoserver/geoserver/graphs/contributors) and commercial [support providers](/support) available to work on GeoServer roadmap items.
+
+## Spring Framework 6 Tasks
+
+The key challenge we are building towards is a spring-framework 6 update, ideally by the end of 2024 when the version we use now reaches end-of-life.
+
+The tasks below are steps towards this goal.
+
+### Wicket 9 upgrade
+
+Interested Parties:
+
+* Brad has been doing amazing work with the Wicket 9 upgrade and is in need of assistance.
+* GeoCat has offered to do manual A/B testing when PR is ready for testing
+
+Activity:
+
+* [[GEOS-11275]](https://osgeo-org.atlassian.net/browse/GEOS-11275) Wicket 9 upgrade
+* [geoserver#7154](https://github.com/geoserver/geoserver/pull/7154)
+
+### Spring Security 5.8 update
+
+Spring Security 5.8 provides a safe stepping stone ahead of the complete spring-framework 6 upgrade and is an activity that can be worked on immediately. 
+
+Interested parties:
+- Andreas (ITS Digital Solutions GmbH) offered to work on this activity in during the initial January call out, and has indicated they are now ready to start.
+- GeoSolutions has indicated they are in position to work on this activity if financing is available
+
+Activity:
+
+* [[GEOS-11271]](https://osgeo-org.atlassian.net/browse/GEOS-11271) Upgrade spring-security to 5.8
+
+### Spring Security Core implementation of OAuth2 / OIDC
+
+The spring-security-oauth client has reached end-of-life and a GeoServer OAuth2 support must be rewritten or migrated as a result.
+
+There are two paths to migrate to spring-security-core implementation:
+
+* Option: Migrate the existing community module implementations to spring-security-core in place; with as little loss of functionality as possible. This has the advantage of using existing test coverage to maintain a consistent set of functionality.
+
+* Option: Setup a community module alongside the existing implementation with the goal of making a full supported etension. This approach has the advantages of allowing organisations the ability to do A/B testing as both the old and new implementation would be available alongside each other.
+
+Security integrations often require infrastructure to develop and test against, which the core GeoServer team does not have access to for automated tests. While the initial priority will be basic OAuth2 and OIDC parties interested in maintaining support for Google, Azure, GeoNode, etc... are welcome to participate.
+
+Interested Parties:
+
+- Andreas (ITS Digital Solutions GmbH) offered to work on this activity and has indicated they are now ready to start
```

Comment:
Did he specifically say anything about the OIDC community module though? Spring security upgrade is one thing, removing the usage of the external, unmaintained OIDC library is something different.
I'd have a statement like "Geosolutions and GeoCat are in a position to do this work..." without dragging other parties in (unless they want to!).

Reply:
He did in an email this week:

> I am now starting to work on:
>  
> GEOS-11271 : Upgrade spring-security to 5.8
> GEOS-11272 : spring-security-oauth replacement, with spring-security 5.8
>  
> As far as I know no activities have taken place in this area so far. Otherwise please let me know.